sdk/python/kfp/dsl/templates/notebook_executor.py

```diff
@@ -0,0 +1,258 @@
+"""Template for notebook executor code generation.
+
+This module contains the actual Python functions that get embedded into notebook
+components at runtime. Using inspect.getsource() to convert them to source code
```

Comment:
Awesome approach. Is it worthwhile doing it for `_generate_shared_extraction_helper` as well?